x-pack/test_serverless/api_integration/test_suites/common/security/sessions.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from 'expect';
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default function ({ getService }: FtrProviderContext) {
+  const svlCommonApi = getService('svlCommonApi');
+  const supertest = getService('supertest');
+
+  describe('security/sessions', function () {
+    describe('route access', () => {
+      describe('disabled', () => {
+        it('invalidate', async () => {
+          const { body, status } = await supertest
+            .post('/api/security/session/_invalidate')
+            .set(svlCommonApi.getInternalRequestHeader())
+            .send({ match: 'all' });
+          svlCommonApi.assertApiNotFound(body, status);
+        });
+      });
+
+      describe('internal', () => {
+        it('get session info', async () => {
+          let body: any;
+          let status: number;
+
+          ({ body, status } = await supertest
+            .get('/internal/security/session')
+            .set(svlCommonApi.getCommonRequestHeader()));
+          // expect a rejection because we're not using the internal header
+          expect(body).toEqual({
+            statusCode: 400,
+            error: 'Bad Request',
+            message: expect.stringContaining(
+              'method [get] exists but is not available with the current configuration'
+            ),
+          });
+          expect(status).toBe(400);
+
+          ({ body, status } = await supertest
+            .get('/internal/security/session')
+            .set(svlCommonApi.getInternalRequestHeader()));
+          // expect 204 because there is no session
+          expect(status).toBe(204);
+        });
+
+        it('extend', async () => {
+          let body: any;
+          let status: number;
+
+          ({ body, status } = await supertest
+            .post('/internal/security/session')
+            .set(svlCommonApi.getCommonRequestHeader()));
+          // expect a rejection because we're not using the internal header
+          expect(body).toEqual({
+            statusCode: 400,
+            error: 'Bad Request',
+            message: expect.stringContaining(
+              'method [post] exists but is not available with the current configuration'
+            ),
+          });
+          expect(status).toBe(400);
+
+          ({ body, status } = await supertest
+            .post('/internal/security/session')
+            .set(svlCommonApi.getInternalRequestHeader()));
+          // expect redirect
+          expect(status).toBe(302);
```

Comment:
The `kbn-xsrf` header is getting set. I've tried also setting a version header, but I still get a 302.

Reply:
Sorry, I just realized that the only thing this API does is always redirecting. :man_facepalming:
```ts
router.post(
    {
      path: '/internal/security/session',
      validate: false,
    },
    async (_context, _request, response) => {
      // We can't easily return updated session info in a single HTTP call, because session data is obtained from
      // the HTTP request, not the response. So the easiest way to facilitate this is to redirect the client to GET
      // the session endpoint after the client's session has been extended.
      return response.redirected({
        headers: {
          location: `${basePath.serverBasePath}/internal/security/session`,
        },
      });
    }
  );
```